scripts/upgrade-react.js

```diff
@@ -0,0 +1,34 @@
+// This file is used to run our nightly cron jobs to test gatsby
+// against future versions of react.
+// @see https://reactjs.org/blog/2019/10/22/react-release-channels.html
+// It updates all of our packages react/react-dom to an experimental
+// channel, for which the circle ci config then installs.
+//
+// This file intentionally does not use in node-packages because it is
+// ran before we do an install.
+const fs = require(`fs`)
+
+const packages = fs.readdirSync(`./packages`)
+
+function replace(deps, library) {
+  if (deps && deps[library]) {
+    deps[library] = process.env.REACT_CHANNEL || `next`
+  }
+}
+
+packages.forEach(packageName => {
```

Comment:
We might have to update/regenerate the monorepo root `yarn.lock` too since dependencies from all workspace packages will be pinned in it and that won't include the update to `next` being done in this script?

Reply:
Absolutely. This is going to happen in a circleci step (the actual yarn install).

Reply:
Wouldn't that be necessary before the `yarn install` in Circle? Since this `yarn.lock` file is committed. Or are we deleting the lock file before running `yarn install`?

Reply:
I don't believe so, and my tests have further proven that this works. e.g. see the screenshot below for the react version:

![image](https://user-images.githubusercontent.com/2916023/68807562-a9c86900-062d-11ea-95e9-9bc74f902cd4.png)


I'm pretty sure yarn.lock is only respected if the package.json versions don't change. If you change a version, running yarn install should install the new version and update the yarn.lock

Reply:
Interesting. `install_node_modules` calls `yarn --frozen-lockfile` which if I remember correctly will _fail_ if an update to the lockfile is necessary 🤔

Reply:
correct, i'm not calling `install_node_modules`, i'm just calling `yarn install` here

Reply:
Here is an example run you can look at. https://circleci.com/workflow-run/d610a002-4067-43f4-97a4-76d006e0681e

note: the failing test here was intentionally failing to test slack hooks